### ISSUE TITLE
Retrieval Eval error for KGs with no object properties 

### ISSUE THREAD

**Issue by LckyLke** [dice-group/Ontolearn#487]
https://github.com/dice-group/Ontolearn/blob/90996aa453d1ab3526f308f8ee24eae447cfa26d/examples/retrieval_eval.py#L89-L91

This causes in error if we use a KG with no object properties. (e.g. lymphography.owl)
Thus we should change the maximum to `max(0, int(len(nc) * args.ratio_sample_nc)`.

**Comment by Demirrr** [comment]
Yes. Could you fix it ?

**Comment by LckyLke** [comment]
> Yes. Could you fix it ?

Yes! Ig the same holds for this sampling of named concepts:
```py
    if args.ratio_sample_nc:
        # (6.1) Subsample if required.
        nc = {i for i in random.sample(population=list(nc), k=max(1, int(len(nc) * args.ratio_sample_nc)))}
```
but are there kgs that dont define classes?

**Comment by Demirrr** [comment]
Yes!

**Comment by LckyLke** [comment]
ok then i will change that as well :)

**Comment by Demirrr** [comment]
Thank you 👍

**Comment by Demirrr** [comment]
fixed in #488
